### PR TITLE
added exception handling that were causing crash and added reinit and…

### PR DESCRIPTION
**Pull request by Sajidi**
@mohhamednabil   please have look on it.

### CODE REVIEW

File: app/src/main/java/com/usepace/android/messagingcenter/clients/connection_client/ClientInterface.java [hunk 1]
```diff
@@ -20,6 +20,8 @@
 abstract class ClientInterface {
 
     abstract public void connect(Context context, ConnectionRequest connectionRequest, ConnectionInterface connectionInterface);
+    abstract public void reConnect();
```
Comment (mohhamednabil) [inline app/src/main/java/com/usepace/android/messagingcenter/clients/connection_client/ClientInterface.java]: it shouldn't be accessible in the app level

File: app/src/main/java/com/usepace/android/messagingcenter/clients/connection_client/ClientInterface.java [hunk 2]
```diff
@@ -20,6 +20,8 @@
...
+    abstract public void reInit(Context context);
```
Comment (mohhamednabil) [inline app/src/main/java/com/usepace/android/messagingcenter/clients/connection_client/ClientInterface.java]: same here

File: app/src/main/java/com/usepace/android/messagingcenter/screens/sendbird/SendBirdChatActivity.java [hunk 3]
```diff
@@ -192,12 +192,19 @@ public void finish() {
     protected void onDestroy() {
         setResult(MessageCenter.OPEN_CHAT_VIEW_RESPONSE_CODE);
         super.onDestroy();
-        SendBird.disconnect(new SendBird.DisconnectHandler() {
```
Comment (mohhamednabil) [inline app/src/main/java/com/usepace/android/messagingcenter/screens/sendbird/SendBirdChatActivity.java]: there is a need for it to enable notifications when the user close the chat view, Make sure you test this edge case

Comment (Sajidi) [reply app/src/main/java/com/usepace/android/messagingcenter/screens/sendbird/SendBirdChatActivity.java]: I have updated and pushed the code

File: app/src/main/java/com/usepace/android/messagingcenter/clients/connection_client/ClientInterface.java (hunk 1 above)
Comment (Sajidi) [reply app/src/main/java/com/usepace/android/messagingcenter/clients/connection_client/ClientInterface.java]: It is already not accessible to outside app, and I have just added 2 new methods to it.

File: app/src/main/java/com/usepace/android/messagingcenter/clients/connection_client/ClientInterface.java (hunk 2 above)
Comment (Sajidi) [reply app/src/main/java/com/usepace/android/messagingcenter/clients/connection_client/ClientInterface.java]: same here

File: app/src/main/java/com/usepace/android/messagingcenter/screens/sendbird/SendBirdChatFragment.java [hunk 4]
```diff
@@ -988,23 +1044,27 @@ public void onSent(FileMessage fileMessage, SendBirdException e) {
     }
 
     private void editMessage(final BaseMessage message, String editedMessage) {
```
Comment (mohhamednabil) [inline app/src/main/java/com/usepace/android/messagingcenter/screens/sendbird/SendBirdChatFragment.java]: You made changes here although the sdk doesn't support this feature yet

File: app/src/main/java/com/usepace/android/messagingcenter/clients/connection_client/SendBirdClient.java [hunk 5]
```diff
@@ -65,13 +160,22 @@ public void onRegistered(SendBird.PushTokenRegistrationStatus status, SendBirdEx
                                                     if (e != null) {    // Error.
                                                         connectionInterface.onMessageCenterConnectionError(e.getCode(), new MessageCenterException(e.getMessage()));
                                                     } else {
-                                                        SendBird.disconnect(new SendBird.DisconnectHandler() {
```
Comment (mohhamednabil) [inline app/src/main/java/com/usepace/android/messagingcenter/clients/connection_client/SendBirdClient.java]: also this disconnect is needed for notifications

File: app/build.gradle [hunk 6]
```diff
@@ -8,8 +9,8 @@ android {
     defaultConfig {
         minSdkVersion 15
         targetSdkVersion 27
-        versionCode 1
-        versionName "1.0"
+        versionCode 2
+        versionName "1.1"
```
Comment (mohhamednabil) [inline app/build.gradle]: version name and version code are not needed to be updated

File: app/src/main/java/com/usepace/android/messagingcenter/clients/connection_client/MessageCenter.java [hunk 7]
```diff
@@ -44,6 +44,43 @@ public static void connect(Context context, ConnectionRequest connectionRequest,
...
     }
 
+
+    /**
+     *
+     * @param context
+     *
+     *
+     */
+    public static void reInitClient(Context context) {
```
Comment (mohhamednabil) [inline app/src/main/java/com/usepace/android/messagingcenter/clients/connection_client/MessageCenter.java]: should not be accessible by app

File: app/src/main/java/com/usepace/android/messagingcenter/clients/connection_client/MessageCenter.java [hunk 8]
```diff
@@ -44,6 +44,43 @@ public static void connect(Context context, ConnectionRequest connectionRequest,
...
+
+
+
+    /**
+     *
+     *
+     *
+     *
+     */
+    public static void reConnect() {
```
Comment (mohhamednabil) [inline app/src/main/java/com/usepace/android/messagingcenter/clients/connection_client/MessageCenter.java]: should not be accessible by app

File: app/src/main/java/com/usepace/android/messagingcenter/clients/connection_client/SendBirdClient.java (hunk 5 above)
Comment (Sajidi) [reply app/src/main/java/com/usepace/android/messagingcenter/clients/connection_client/SendBirdClient.java]: done

File: app/build.gradle (hunk 6 above)
Comment (Sajidi) [reply app/build.gradle]: reverted

File: app/src/main/java/com/usepace/android/messagingcenter/screens/sendbird/SendBirdChatFragment.java (hunk 4 above)
Comment (Sajidi) [reply app/src/main/java/com/usepace/android/messagingcenter/screens/sendbird/SendBirdChatFragment.java]: did not change anything, might be spaces etc..

File: app/src/main/java/com/usepace/android/messagingcenter/screens/sendbird/SendBirdChatActivity.java (hunk 3 above)
Comment (Sajidi) [reply app/src/main/java/com/usepace/android/messagingcenter/screens/sendbird/SendBirdChatActivity.java]: done

File: app/src/main/java/com/usepace/android/messagingcenter/clients/connection_client/MessageCenter.java (hunk 8 above)
Comment (Sajidi) [reply app/src/main/java/com/usepace/android/messagingcenter/clients/connection_client/MessageCenter.java]: resolved by adding a extension class, as discussed

File: app/src/main/java/com/usepace/android/messagingcenter/clients/connection_client/MessageCenter.java (hunk 7 above)
Comment (Sajidi) [reply app/src/main/java/com/usepace/android/messagingcenter/clients/connection_client/MessageCenter.java]: resolved by adding a new class as discussed
